src/pykoala/plotting/plot_plot.py

```diff
@@ -0,0 +1,418 @@
+        Thickness of the axes
+    save_file : string (default = none)
+        Specify path and filename to save the plot
+    path: string  (default = "")
+        path of the file to be saved
+    fig_size : float (default = 12)
+        Size of the figure
+    warnings : boolean (default = True)
+        Print the warnings in the console if something works incorrectly or might require attention 
+    show : boolean (default = True)
+        Show the plot
+    statistics : boolean (default = False)
+        Print statistics of the data in the console
+       
+   """
+      
+   if fig_size == "big":
+       fig_size=20
+       label_axes_fontsize=20 
+       axes_fontsize=15
+       title_fontsize=22
+       tick_size=[10,1,5,1]
+       axes_thickness =3
+       hwidth =2
+       vwidth =2
+
+   if fig_size in ["very_big", "verybig", "vbig"]:
+       fig_size=35
+       label_axes_fontsize=30 
+       axes_fontsize=25
+       title_fontsize=28 
+       tick_size=[15,2,8,2]
+       axes_thickness =3
+       hwidth =4 
+       vwidth =4
+
+   if fig_size != 0 : 
+       if type(fig_size) == list:
+           plt.figure(figsize=(fig_size[0], fig_size[1]))
+       else:     
+           plt.figure(figsize=(fig_size, fig_size/2.5))
+   
+   if np.isscalar(x[0]) :
+       xx=[]
+       for i in range(len(y)):
+           xx.append(x)
+   else:
+       xx=x
+
+   if xmin is None : xmin = np.nanmin(xx[0])
+   if xmax is None : xmax = np.nanmax(xx[0])  
+   
+   alpha_=alpha
+   psym_=psym
+   label_=label
+   linewidth_=linewidth
+   markersize_=markersize
+   linestyle_=linestyle
+
+   n_plots=len(y)
+       
+   if np.isscalar(y[0]) ==  False:
+       if np.isscalar(alpha):        
+           if alpha_ == "":
+               alpha =[0.5]*n_plots
+           else:
+               alpha=[alpha_]*n_plots
+       if np.isscalar(psym): psym=[psym_]*n_plots
+       if np.isscalar(label): label=[label_]*n_plots
+       if np.isscalar(linewidth): linewidth=[linewidth_]*n_plots
+       if np.isscalar(markersize):markersize=[markersize_]*n_plots
+       if np.isscalar(linestyle): linestyle=[linestyle_]*n_plots
+       if color == "blue" : 
+           color_list = ["red","blue","green","k","orange", "purple", "cyan", "lime"]
+           color = []
+           j=0
+           for i in range(n_plots):
+               color.append(color_list[j])
+               j = j+1
+               if j == 8 : j = 0
+           
+           #color = [color_list[i] for i in range(n_plots)]
+       if ymax is None: y_max_list = []
+       if ymin is None: y_min_list = []
+              
+       if fcal is None:
+           if np.nanmedian(np.abs(y[0])) < 1E-10:
+               fcal = True
+               if np.nanmedian(y[0]) < 1E-20 and np.var(y[0]) > 0 : fcal = False
+       for i in range(len(y)):
+           if psym[i] == "":
+               plt.plot(xx[i],y[i], color=color[i], alpha=alpha[i], label=label[i], linewidth=linewidth[i], linestyle=linestyle[i])
```

Comment:
Why do you need a for loop, you can pass two lists as x, y arguments to `plt.plot`, this makes this method extremely inefficient...

Reply:
I am not sure. I think @angelrls 's idea is that you can tell the routine to plot one spectrum or a list of spectra (with common or individual wavelength). That's why `x` and `y` may be single objects or lists. If so, every spectrum (element of `y`) is expected to have its own color, alpha, etc. Is this correct? I'm not sure what's the best way to handle this, but the loop is probably not too terrible, as any plot would arguably contain a small number of spectra...

Reply:
x and y can be just a list, plot_plot(wave, intensity), or y can be a list of lists, `plot_plot (wave, [intensity[0], intensity[1]])` or two lists of lists, `plot_plot ([wave1, wave2], [[intensity[0], intensity[1]]) `. Yes, each element can have their own color, alpha, etc

 I'm sure there are better ways of coding it, that is why I said that arch could be improved, but it works and in any case it is expected that this will only use few inputs (<10, and typically 2-4 at max).

I particularly find it extremely useful as I don't have to check how to put minor axes, change colours, style, alpha... etc, just check the "help" of plot_plot and go. I had a similar thing for my plots in IDL.